src/harness/testcases/WINNF_FT_S_WDB_testcase.py

```diff
@@ -31,28 +33,133 @@ def setUp(self):
+    by invoking the PPA creation status API.If the status is complete then looks for
+    withError set to True to declare ppa creation is failed. The status is checked
+    every 10 secs for up to 2 hours.
+
+    Args:
+      ppa_creation_request: A dictionary with a multiple key-value pair containing the
+        "cbsdIds", "palIds" and optional "providedContour"(a GeoJSON object).
+
+    Raises:
+      Exception for the following cases:
+         a. withError flag set to False even though most recent ppa creation
+         status is completed.
+         b. SAS UUT does not respond for longer time and timeout.
+         c. SAS UUT responds with success response rather than failure response.
+
+    """
+    response = self._sas_admin.TriggerPpaCreation(ppa_creation_request)
+    logging.info('TriggerPpaCreation is in progress')
+
+    # Triggers most recent PPA Creation Status immediately and checks for the status
+    # of activity every 10 seconds until it is completed. If the status is not changed
+    # within 2 hours it will throw an exception.
+    signal.signal(signal.SIGALRM,
+                  lambda signum, frame:
+                  (_ for _ in ()).throw(
+                      Exception('Most Recent PPA Creation Status Check Timeout')))
+
+    # Timeout after 2 hours if it's not completed.
+    signal.alarm(7200)
+
+    # Check the Status of most recent ppa creation every 10 seconds.
+    while not self._sas_admin.GetPpaCreationStatus()['completed']:
+      time.sleep(10)
+
+    # Expect the withError flag in status response should be toggled on to True
+    # to indicate PPA creation failure.
+    self.assertTrue(self._sas_admin.GetPpaCreationStatus()['withError'],
+                    msg='Expected:There is an error in create PPA. But '
+                        'PPA creation status indicates no error')
+    signal.alarm(0)
+
   def generate_WDB_1_default_config(self, filename):
     """Generates the WinnForum configuration for WDB.1"""
 
-    # Load device info
+    # Load devices info
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
     device_b = json.load(
-      open(os.path.join('testcases', 'testdata', 'device_b.json')))
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
 
-    # Load PAL database record
-    pal_record_0 = json.load(
+    # Load PAL records
+    pal_record_a_0 = json.load(
       open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    pal_record_a_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_1.json')))
 
-    # Fake database test harness configuration
-    # The databaseFile WDB_1_PAL_Database.csv is populated with PAL record 
-    # details of pal_record_0
-    fake_pal_database_config = {
-        'hostName': 'localhost',
-        'port': 9090,
-        'databaseFile': os.path.join('testcases', 'testdata', 'wdb_1', 'WDB_1_PAL_Database.csv')
-    }
+    pal_record_b_0 = json.load(
```

Comment:
I don't think you want to use the same PAL records for both tests, otherwise you risk overlap/interaction between the test cases (see my comment below).

Reply:
Updated,   used different PAL records